microsetta_admin/server.py

```diff
@@ -401,63 +402,80 @@ def per_sample_summary():
-            # there was an error. abort early.
-            return render_template('per_sample_summary.html',
-                                   resource=None,
-                                   projects=projects,
-                                   **build_login_variables(),
-                                   search_error=[{'error': err}])
+        search_field = request.form.get('search_field')
+        search_value = request.form.get('text_input')
+        uploaded_file = request.files.get('file')
 
-    # perform the main query.
-    payload = {'sample_barcodes': sample_barcodes}
-    status, result = APIRequest.post('/api/admin/account_barcode_summary?stri'
-                                     'p_sampleid=%s' % str(strip_sampleid),
-                                     json=payload)
+        search_values = []
 
-    if status == 200:
-        if result['partial_result'] is True:
-            unprocessed_barcodes = result['unprocessed_barcodes']
+        if uploaded_file:
+            file_content = io.TextIOWrapper(uploaded_file,
+                                            encoding='utf-8-sig')
+            csv_reader = csv.reader(file_content)
+
+            for row in csv_reader:
+                search_values.extend(row)
         else:
-            unprocessed_barcodes = None
-        resource = pd.DataFrame(result['samples'])
-        order = ['sampleid', 'project', 'account-email',
-                 'source-type', 'site-sampled', 'sample-date',
-                 'sample-time', 'sample-status', 'sample-received',
-                 'ffq-taken', 'ffq-complete', 'vioscreen_username']
-        order.extend(sorted(set(resource.columns) - set(order)))
-        resource = resource[order]
+            search_values = [search_value] if search_value else []
+
+        payload = {}
+        if search_field == 'sample_barcode':
+            payload['sample_barcodes'] = search_values
+        elif search_field == 'kit_id':
+            payload['kit_ids'] = search_values
+        elif search_field == 'email':
+            payload['emails'] = search_values
+        elif search_field == 'outbound_tracking':
+            payload['outbound_tracking_numbers'] = search_values
+        elif search_field == 'inbound_tracking':
+            payload['inbound_tracking_numbers'] = search_values
+
+        # perform the main query.
+        status, result = APIRequest.post('/api/admin/account_barcode_summary?'
+                                         'strip_sampleid=%s' %
+                                         str(strip_sampleid),
+                                         json=payload)
+
+        if status == 200:
+            if result['partial_result'] is True:
+                unprocessed_barcodes = result['unprocessed_barcodes']
+            else:
+                unprocessed_barcodes = None
+            resource = pd.DataFrame(result['samples'])
+            if not resource.empty:
+                order = ['sampleid', 'project', 'account-email',
```

Comment:
Can you please extend this to cover all of the columns being returned in the report so that they're logically grouped? If you're unsure about how to order any of the new columns, please let me know.

Reply:
I added the additional columns and grouped them how I saw fit, let me know if it needs adjusting.